packages/x-tailwindcss/demo/src/components/xds-button.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <div class="x-flex x-flex-col x-gap-16">
+    <h1 class="x-text-lg">Button</h1>
+    <div
+      v-for="(classes, section) in sections"
+      :key="section"
+      class="x-flex x-flex-row x-items-start x-gap-12 x-align-items-baseline"
+    >
+      <h2 class="x-text-md">{{ section }}</h2>
+      <button v-for="cssClass in classes" :key="cssClass" @click="copyCSSClasses" :class="cssClass">
+        {{ removeBase(cssClass) }}
+      </button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  import { Vue, Component, Prop } from 'vue-property-decorator';
+
+  @Component
+  export default class XdsButtonShowcase extends Vue {
```

Comment:
I'm wondering whether it makes sense or not to have a mixin covering all the logic that may be shared between different showcase components in the future. In the end I understand that sizes, colors, etc. apply to most of them, so my suggestion is something like this:
```

import { Vue, Component, Prop } from 'vue-property-decorator';

@Component
export default class XdsComponentShowcaseMixin extends Vue {
  @Prop({ required: true })
  public baseClass!: string;

  @Prop({ default: () => ['xs', 'sm', 'md', 'lg', 'xl'] })
  public sizes!: string[];

  @Prop({
    default: () => [
      'neutral',
      'lead',
      'auxiliary',
      'accent',
      'highlight',
      'success',
      'warning',
      'error'
    ]
  })
  public colors!: string[];

  @Prop()
  public combinations!: string[];

  protected get sections(): Record<string, string[]> {
    return {
      Default: [this.baseClass],
      Colors: this.colors.map(this.prefixWith(this.baseClass)),
      Sizes: this.sizes.map(this.prefixWith(this.baseClass)),
      ...(this.combinations && {
        Combinations: this.combinations.map(this.prefixWith(this.baseClass))
      })
    };
  }

  protected prefixWith(prefix: string): (value: string) => string {
    return cssClass => `${prefix} ${prefix}-${cssClass}`;
  }
}

```

Reply:
I think @alvarodE comment makes sense. Another option is to create a generic component that receives via props the configuration. The `showcase` component could get the tailwind config via [resolveConfig](https://tailwindcss.com/docs/configuration#referencing-in-java-script) so we have the prefix and colors. Afterwards a generic `xds-component` could just paint the given config. Something like 

```
// Showcase component
    mounted(): void {
      const config = resolveConfig(this.tailwindConfigFile);
      this.colors = config.theme?.colors ?? {};
      this.prefix = config.prefix ??  '';
    }
```

```<template>
  <div>
    <h3>{{ componentConfig.tag }}</h3>
    <ul>
      <li v-for="(_, colorName) in tailwindConfig.colors" :key="colorName">
        <component
          :is="componentConfig.tag"
          :class="[
            `${tailwindConfig.prefix}${componentConfig.base}-${colorName}`,
            componentConfig.classes
          ]"
        >
          {{ colorName }}
        </component>
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
  import { Component, Prop, Vue } from 'vue-property-decorator';
  @Component
  export default class XdsItem extends Vue {
    @Prop({
      required: true
    })
    public tailwindConfig!: {
      prefix: string;
      colors: Record<string, object>;
    };
    @Prop({
      required: true
    })
    public componentConfig!: {
      tag: string;
      base: string;
      classes: string;
    };
  }
</script>
```


In the example above only colors are displayed but I think the point is clear 😅

Reply:
I thought about 2 approaches. But it is something I decided not to do in this PR because of early abstractions. I don't have a clear visibility of what would be helpful for each component that we design. So until we have 2-3 more components I wouldn't abstract the utilities to generate this showcase.